app/controllers/posts_controller.rb

```diff
@@ -31,6 +33,16 @@ def create
     end
   end
 
+  def destroy
+    post = Post.find(params[:id])
+    user = User.find_by(id: post.author_id)
+    user.posts_counter -= 1
+    post.destroy
+    user.save
+    redirect_to user_posts_path(user.id)
+    flash[:success] = 'Post Deleted Successfully'
+  end
+
```

Comment:
hi @DeliceLydia thanks for the review the destroy action works but it works when there are no likes or comments because that will raise a foreign key violation

Reply:
Ok I fixed it, thanks.